### PR TITLE
[JUJU-1971] Fix namespace for k8s controller model

### PR DESCRIPTION
**Pull request by barrettj12**
```console
$ microk8s enable rbac
$ juju bootstrap microk8s k1
$ juju ssh --debug -m k1:controller 0
ERROR unable to determine legacy status for namespace controller: namespaces "controller" is forbidden: User "system:serviceaccount:controller-k1:model-exec" cannot get resource "namespaces" in API group "" in the namespace "controller"
```

The issue is here
https://github.com/juju/juju/blob/5296f9369f8efcfb21bf003fd8bb82917d0ec833/caas/kubernetes/provider/exec/exec.go#L98-L105
The first argument to `k8sutils.IsLegacyModelLabels` should be the namespace (= `controller-k1`), but instead we are providing the model name (= `controller`).

There is a method `modelNameToNameSpace` to get the namespace from the model name, but this requires knowing if the model has legacy labels, so it's a catch 22.

My fix is to just pass the namespace as an argument to `NewForJujuCloudSpec`. This removes all the convoluted logic above and saves us from making extra k8s api calls.

It is easy to work out the namespace from the model name, but for the `controller` model, we need to know the controller name too. In this case, we use the controller API to determine the model name from controller config.

As a drive-by, I've fixed several typos in comments and error messages.

## TODO
- [x] add a test
- [x] clean up git history

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] ~Comments saying why design decisions were made~
- [x] Go unit tests, with comments saying what you're testing
- [x] ~[Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- [x] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```console
$ microk8s enable rbac
$ juju bootstrap microk8s k1
$ juju ssh --debug -m k1:controller 0
```

## Bug reference
https://bugs.launchpad.net/juju/+bug/1991094

### CODE REVIEW

**Review by tlm — APPROVED** [review]
lgtm just need to bump the facade version as discussed.

File: cmd/juju/commands/ssh_container.go [hunk 1]
```diff
@@ -450,6 +451,12 @@ func (c *sshContainer) getCloudSpec() (cloudspec.CloudSpec, error) {
 	}
 	c.modelName = mInfo.Result.Name
```
Comment (ycliuhw) [inline cmd/juju/commands/ssh_container.go:453]: Code below line 443 are legacy. We should find the controller name before line 439(probably just do it in `initRun` or `resolveTarget`).

File: apiserver/facades/client/client/register.go [hunk 2]
```diff
@@ -25,6 +26,9 @@ func Register(registry facade.FacadeRegistry) {
 		return newFacadeV4(ctx)
 	}, reflect.TypeOf((*ClientV4)(nil)))
 	registry.MustRegister("Client", 5, func(ctx facade.Context) (facade.Facade, error) {
+		return newFacadeV5(ctx)
+	}, reflect.TypeOf((*Client)(nil)))
+	registry.MustRegister("Client", 6, func(ctx facade.Context) (facade.Facade, error) {
```
Comment (ycliuhw) [inline apiserver/facades/client/client/register.go:31]: We need to bump version in `api/facadeversions.go` as well.

File: cmd/juju/commands/ssh_container_test.go [hunk 3]
```diff
@@ -824,3 +830,48 @@ func (s *sshContainerSuite) TestCopyToWorkloadPodWithContainerSpecified(c *gc.C)
...
+
+func (f *fakeModelCommand) NewAPIClient() (*apiclient.Client, error) {
+	return nil, nil
+}
+
+func (f *fakeModelCommand) ModelIdentifier() (string, error) {
+	return f.modelName, nil
+}
+
+type fakeControllerAPI struct {
```
Comment (ycliuhw) [inline cmd/juju/commands/ssh_container_test.go:871]: Given you have already setup gomock, why do we need this fake API?

File: cmd/juju/commands/ssh_container_test.go [hunk 4]
```diff
@@ -824,3 +830,48 @@ func (s *sshContainerSuite) TestCopyToWorkloadPodWithContainerSpecified(c *gc.C)
...
+	s.controllerAPI = &fakeControllerAPI{"foobar"}
+	ctrl := s.setUpController(c, true, "")
+	defer ctrl.Finish()
+
+	err := s.sshC.InitRun(&mc)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(s.sshC.Namespace(), gc.Equals, "controller-foobar")
+}
+
+type fakeModelCommand struct {
```
Comment (ycliuhw) [inline cmd/juju/commands/ssh_container_test.go:847]: We can just use gomock as well

**Review by ycliuhw — APPROVED** [review]
LGTM, just need to update the test.
Thank you for fixing this.

**Pull request merged**